cluster-health.go

```diff
@@ -141,44 +141,53 @@ func (an *AnonymousClient) Alive(ctx context.Context, opts AliveOpts, servers ..
 	resultsCh = make(chan AliveResult)
 	go func() {
 		defer close(resultsCh)
-		for _, server := range servers {
-			u, err := url.Parse(an.endpointURL.Scheme + "://" + server.Endpoint)
-			if err != nil {
-				resultsCh <- AliveResult{
-					Error: err,
-				}
-				return
-			}
-			t := time.Now()
-			resp, err := an.executeMethod(ctx, http.MethodGet, requestData{
-				relPath:          resource,
-				endpointOverride: u,
-			})
-			responseTime := time.Since(t)
-			closeResponse(resp)
-
-			var result AliveResult
-			if err != nil {
-				result = AliveResult{
-					Endpoint:     u,
-					Error:        err,
-					ResponseTime: responseTime,
-				}
-			} else {
-				result = AliveResult{
-					Endpoint:     u,
-					ResponseTime: responseTime,
-					Online:       resp.StatusCode == http.StatusOK && resp.Header.Get("x-minio-server-status") != "offline",
+		if len(servers) == 0 {
+			an.alive(ctx, an.endpointURL, resource, resultsCh)
+		} else {
+			for _, server := range servers {
+				u, err := url.Parse(an.endpointURL.Scheme + "://" + server.Endpoint)
+				if err != nil {
+					resultsCh <- AliveResult{
+						Error: err,
+					}
+					return
```

Comment:
Shall we continue here instead of returning ?

Reply:
done.